ebean-core/src/main/java/io/ebeaninternal/server/query/CQueryBuilder.java

```diff
@@ -585,16 +585,11 @@ private void appendSelect() {
         }
         if (distinct && dbOrderBy != null) {
           // add the orderBy columns to the select clause (due to distinct)
-          final OrderBy<?> orderBy = query.getOrderBy();
-          if (orderBy != null && orderBy.supportsSelect()) {
-            String trimmed = DbOrderByTrim.trim(dbOrderBy);
-            if (query.isSingleAttribute() && select.selectSql().startsWith(trimmed)) {
-              // NOP, already in SQL
-              // TODO: what to do if we select("id").orderBy("prop,id")?
-              // Can we live with a query like "select t0.id, t0.prop, t0.id from"
-              // or should we elliminate the second "t0.id" from select
-            } else {
-              sb.append(", ").append(trimmed);
+          String[] tokens = DbOrderByTrim.trim(dbOrderBy).split(",");
```

Comment:
here, we do not longer check `query.getOrderBy()`, but we check only `dbOrderBy` which is effectively added also by the old code. It seems, that `supportsSelect` is no longer used.

> Can we live with a query like "select t0.id, t0.prop, t0.id from or should we elliminate the second "t0.id" from select

the new code adds only the tokens, that are not yet part of the `selectSql`